### PR TITLE
Update windows build instructions.

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
Fixes https://github.com/grpc/grpc/issues/13509

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @SurferJeffAtGoogle isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
The Linux Foundation CLA GitHub bot_

**Comment by SurferJeffAtGoogle** [comment]
I believe I'm covered by the Google corporate CLA.  I am an employee of Google.

**Comment by ctiller** [comment]
Can anything be done to simplify these instructions? (it's getting to be a lot of steps)

**Comment by SurferJeffAtGoogle** [comment]
We could put the build commands (but not the install commands) in a batch file or powershell script.
But I didn't find the steps hard to follow.  And I liked seeing each command one by one so I could understand the dependencies and where the build output would be generated.

This PR is not intended to otherwise improve the build experience.  The issue this PR fixes is that current steps are incomplete and therefore broken.  Let's fix them first, then simplify them.

**Comment by ctiller** [comment]
Yeah... apologies for the meta comment: I think this PR is fine (although I'd like @jtattermusch to be the actual reviewer). I have a general concern that our build instructions are getting more complicated (not less) over time, and this is something we as a team need to start addressing.

**Comment by mehrdada** [comment]
@SurferJeffAtGoogle for the CLA you still need to follow the instructions so that The Linux Foundation Bot pairs with both GitHub and your Google Account (use your corp account there) and can pick you up as a Googler.

**Comment by SurferJeffAtGoogle** [comment]
I singed the CLA.

File: INSTALL.md [hunk 1]
```diff
@@ -106,11 +107,15 @@ Builds gRPC C and C++ with boringssl.
 Please note that when using Ninja, you'll still need Visual C++ (part of Visual Studio)
 installed to be able to compile the C/C++ sources.
 ```
+> powershell git clone -b ((New-Object System.Net.WebClient).DownloadString(\"https://grpc.io/release\").Trim()) https://github.com/grpc/grpc
```
Comment (mehrdada) [inline INSTALL.md]: I suggest adding `--recursive` to the initial `git clone` command instead of a separate `git submodule update --init` statement.

Comment (SurferJeffAtGoogle) [reply INSTALL.md]: Done.

**Comment by mehrdada** [comment]
LGTM, but I leave it to @jtattermusch for final Approval and merge.

**Comment by jtattermusch** [comment]
@ctiller   to answer the question "can we make this simpler":
I'm trying to, here's some analysis:
In terms of prerequisites, we currently have these:
- git, cmake, visual studio - I think those are very reasonable and we can't be really expected to build without them (assuming we are using cmake which IMHO is our best option currently)
- ninja (is optional, you need it if you want faster build or boringssl asm optimizations)
- perl, go, yasm - those actually are all purely boringssl dependencies - so yes, we can get rid of those dependencies if we don't build with boringssl (but in that case you would need to install openssl instead) 
- chocolatey - assuming you want to install perl go and yasm, using choco is the easiest ways I know of currently. So if we didn't need to build boringssl, you don't need chocolatey.
So most of the dependencies are actually required by boringssl, not by grpc code.

In terms of build steps, we have these (in case you choose the easier VS-only way)
- recursive git clone (technically not part of the build but it turns out users don't realize you need that, so 
it probably makes sense to include it)
- create the directory for building and enter it  `md .build` and `cd .build` - not really a step that has much intellectual overhead and it's good for ensuring users are in the right directory.
- cmake "generate projects step" `cmake .. -G "Visual Studio 14 2015" -DCMAKE_BUILD_TYPE=Release`
- actually build using `cmake --build .`
So overall, the actual build steps are already quite minimal.

What can be done
- a bit of docs polishing (to make the minimal resistance path more obvious)
- we could add a .bat script that will do some of the steps for you (but frankly, it's 3 commands not counting cd and md after you installed the dependencies, so I don't know if having a script doesn't add more complexity that what it solves).
- we could investigate how to build without boringssl (that's entirely possible, but building with a different security library than we use on linux is not very good for consistency)

**Comment by jtattermusch** [comment]
Thanks for the contribution!
